### PR TITLE
UI Changes: Add Window Title to Screenshot

### PR DESCRIPTION
**Pull request by StefanGreve**
I added (back?) the filename input feature and managed to center the title somewhat reliably, here are some screenshots:

## macOS
image

## Windows
image

## Changes
- update packages
- lift state up from the `Screenshot` component
- store `fileName` state in `App.tsx`
- pass on `fileName` as `prop` to `CodeWindow`, then one more level deeper to `TitleBar`
- add `.title` class in `style.scss` (of `TitleBar`) as well as a new `div` element to `result`

## Technical Limitations
- long text is still problematic, but there's just no way to properly display a let's say 200-character long title on a small div element without causing an overflow, maybe we should truncate the title or just let it be? I will leave that decision up to you

### CODE REVIEW

**Comment by flyingsl0ths** [comment]
> * long text is still problematic, but there's just no way to properly display a let's say 200-character long title on a small div element without causing an overflow, maybe we should truncate the title or just let it be? I will leave that decision up to you

I think an 11 character limit suffixed by ellipsis would be fine: "Hello World App.js" -> "Hello World..."

File: src/App.tsx [hunk 1]
```diff
@@ -58,6 +60,11 @@ export default class App extends React.Component<
     };
 
     updateTheme(defaultTheme);
+    this.onFileNameChangeHandler = this.onFileNameChangeHandler.bind(this);
```
Comment (flyingsl0ths) [inline src/App.tsx]: Could you make this a private member and rename it to `handleFileNameChange` via class field syntax? I'm aware of the naming conventions for props that represent events and values that handle them I'll be making the changes where necessary soon

Comment (StefanGreve) [reply src/App.tsx]: I used an arrow function here, hope this works as well for you

**Comment by StefanGreve** [comment]
> > * long text is still problematic, but there's just no way to properly display a let's say 200-character long title on a small div element without causing an overflow, maybe we should truncate the title or just let it be? I will leave that decision up to you
> 
> I think an 11 character limit suffixed by ellipsis would be fine: "Hello World App.js" -> "Hello World..."

I actually think that 20 is a more suitable character limit, it is still sufficiently small enough to fit on most cell phone devices

**Pull request merged**